### DISCUSSION TITLE
Issues with $titleColumnKey shifting an extra column to table

### DISCUSSION THREAD

**Discussion by zedee** [area17/twill#1008] [Q&A] [answered]
So I'm trying to display a module which has two relationships.

Got a Crews table which contains members and roles and got a module for it. Maybe not the smartest decision but this was inherited, so... I got what I got.

Part of my controller:

```
protected $moduleName = 'crews';

    protected $formWith = ['member', 'role'];

    protected $titleColumnKey = 'memberName';

    protected $indexOptions = [
        'create' => true,
        'editInModal' => true,
    ];

    protected $indexColumns = [
        'member' => [
            'title' => 'Name',
            'relationship' => 'member',
            'field' => 'name',
            'sort' => true
        ],
        'role' => [ // relation column
            'title' => 'Role',
            'relationship' => 'role',
            'field' => 'title',
            'sort' => true
        ]
    ];

    protected $browserColumns = [
        'member' => [
            'title' => 'Name',
            'relationship' => 'member',
            'field' => 'name'
        ],
        'role' => [ // relation column
            'title' => 'Role',
            //'sort' => true,
            'relationship' => 'role',
            'field' => 'title'
        ]
    ];

    // Optional, specify the default listing order
    protected $defaultOrders = ['member' => 'desc'];

    // Optional, specify the default listing filters
    protected $defaultFilters = ['search' => 'member|search'];
```

On index view, I am being able to get names and roles, but there's an extra column I cannot reach, which defaults to 'name' as its name, and its not sortable (funny, the column next to it, is).

I attach an screenshot to clarify what I got:

![image](https://user-images.githubusercontent.com/5777239/63448764-ef1a3780-c43e-11e9-8d93-1701f0bc7cd2.png)

Any ideas?

**Comment by zedee** [comment]
Has anyone found an explanation for this issue?

**Comment by ifox** [comment]
Hi @zedee sorry for the delay. I'm not sure how you are trying to access the "Nom" column since it is not in your example code. I think you might have an issue because your `titleColumnKey` doesn't have an entry in the `$indexColumns`. Try renaming your `member` key to `memberName`. I'm afraid this is going to be a problem to create new records though, is there no field on the crew model that you'd display on the listing page/create modal?

**Comment by galingong** [comment, marked as answer]
Had the same problem until I figured it out - @ifox is right, I can confirm that if you override `$indexColumns`, `$titleColumnKey` has to have a key in it. Twill tries to access $indexColumns[$this->titleColumnKey], which means that whatever `$titleColumnKey` property is set to, that value has to be included in `$indexColumns` as a key, eg. if you left `$titleColumnKey` untouched (default value is 'title', already set in the package), `$indexColumns` has to be set like so:
```
    protected $indexColumns = [
        'title' => [
            'title' => 'Title',
            'field' => 'title'
        ],
        'member' => [
            'title' => 'Name',
            'relationship' => 'member',
            'field' => 'name',
            'sort' => true
        ],
        'role' => [ // relation column
            'title' => 'Role',
            'relationship' => 'role',
            'field' => 'title',
            'sort' => true
        ]
    ];
```
